presto-docs/src/main/sphinx/admin/resource-groups.rst

```diff
@@ -9,287 +9,410 @@ Except for the limit on queued queries, when a resource group runs out of a reso
 
-* ``softMemoryLimit`` (required): maximum amount of distributed memory this
-  group may use before new queries become queued. May be specified as
-  an absolute value (i.e. ``1GB``) or as a percentage (i.e. ``10%``) of the cluster's memory.
+File-Based Resource Management
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-* ``softCpuLimit`` (optional): maximum amount of CPU time this
-  group may use in a period (see ``cpuQuotaPeriod``) before a penalty will be applied to
-  the maximum number of running queries. ``hardCpuLimit`` must also be specified.
+In a file-based resource manager, configuration information about the resource groups is stored in a JSON file.
+This file contains definitions of all resource groups and the rules to select them.
+The configuration file is loaded and used at the start of the Presto server.
+Any changes to the file require a restart of the Presto server to take effect.
+
+Database-Based Resource Management
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In a database-based resource manager, configuration information about the resource groups is stored in a relational
+database. The database contains definitions of all resource groups and the rules to select them.
+Unlike the file-based resource manager, changes to the configuration in the database take effect
+immediately and do not require a restart of the Presto server.
+
+Both methods have their pros and cons. File-based resource management is simpler to set up but less flexible,
+while database-based resource management is more complex to set up but offers more flexibility and dynamic changes.
+
+File Resource Group Manager
+---------------------------
+
+The file resource group manager in PrestoDB is a way to manage resources using a JSON configuration file.
+This file contains definitions of all resource groups and rules for selecting the appropriate resource group
+for a given query.
+
+To set up a file-based resource group manager:
+
+1. Create the file ``etc/resource-groups.properties``.
+
+2. Set the ``resource-groups.configuration-manager`` property to ``file`` using the following code example:
+
+   .. code-block:: text
+
+       resource-groups.configuration-manager=file
+       resource-groups.config-file=etc/resource-groups.json
+
+3. Create a JSON file in etc folder named resource-groups.json. This file should contain the definitions
+   for the resource groups. Each resource group can specify things like the maximum memory, the
+   maximum queued queries, and the maximum running queries. For information on creating resource group definitions,
+   see `Resource Group Properties`_. For an example of a resource-groups.json file,
+   see `File Resource Group Manager <resource-groups.html#id2>`_.
+
+4. In the ``etc/resource-groups.properties`` file, add a line that specifies the location of the JSON file.
+   Set the ``resource-groups.config-file`` property to ``<file_path>``, where ``<file_path>`` is the path to
+   the JSON file.
```

Comment:
After this line, please add here the one-line code block containing the line

``resource-groups.config-file=<file-path>``

Please make sure that this code block is indented correctly to keep it as part of step 4, the same as you indented the code block in step 2 (lines 71-74).